app/src/main/java/com/github/se/project/model/lesson/LessonRepositoryFirestore.kt

```diff
@@ -0,0 +1,141 @@
+package com.android.sample.model.lesson
+
+import android.util.Log
+import com.google.android.gms.tasks.Task
+import com.google.firebase.firestore.DocumentSnapshot
+import com.google.firebase.firestore.FirebaseFirestore
+
+class LessonRepositoryFirestore(private val db: FirebaseFirestore) : LessonRepository {
+
+  private val collectionPath = "lessons"
+
+  // Initialize the repository
+  override fun init(onSuccess: () -> Unit) {
+    // Additional initialization logic if required
+    onSuccess()
+  }
+
+  // Retrieve all lessons for a specific user by userUid
+  override fun getLessonsByUserId(
+      userUid: String,
+      onSuccess: (List<Lesson>) -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    db.collection("users")
+        .document(userUid)
+        .collection(collectionPath)
+        .get()
+        .addOnCompleteListener { task ->
+          if (task.isSuccessful) {
```

Comment:
After todays meeting, we decided to split the lessons from the profile. Please adapt the whole implementation for that purpose. I mention it once for the whole file. Structure the lessons so that they have everything from the creation of a lesson and a userId, for the user similarly, everything from the profile and a list of lessonsIds.